dbgap_extract.py

```diff
@@ -114,14 +118,29 @@ def main():
             "Usage error. Run this script using one of the two below command forms:"
         )
         logging.debug(
-            "> python dbgapExtract.py --study_accession_list accession_1 accession_2 ...."
+            "> python dbgap_extract.py --study_accession_list accession_1 accession_2 .... [--output_filename file_out.tsv]"
         )
         logging.debug(
-            "> python dbgapExtract.py --study_accession_list_filename file.txt"
+            "> python dbgap_extract.py --study_accession_list_filename file.txt [--output_filename file_out.tsv]"
         )
         logging.debug("-------")
         exit(0)
 
+    output_filename = FILENAME + ".tsv"
+    if args.output_filename is not None:
+        output_filename = args.output_filename
+        # Log to a file matching the filename-prefix supplied by the user.
+        LOG_FILE = output_filename.split(".")[0] + ".log"
+        fileh = logging.FileHandler(LOG_FILE, "a")
+        log = logging.getLogger()
+        for hdlr in log.handlers[:]:
+            if "baseFilename" in dir(hdlr) and FILENAME in hdlr.baseFilename:
```

Comment:
using `dir` in this way is sorta fragile (if they every change things behind the scenes in logging this might break). I'd recommend just removing all the handlers and re creating everything. You could even break out likes 11-14 into a function that you just call here. `setup_logging()` or something (and call it from the top level too)